src/weapon/w_005.c

```diff
@@ -9,7 +9,60 @@ extern s32 g_HandId;
 #define g_Animset w_005_1
 #define g_Animset2 w_005_2
 
-INCLUDE_ASM("weapon/nonmatchings/w_005", EntityWeaponAttack);
+extern WeaponAnimation D_27000_8017B2F4[];
+
+void EntityWeaponAttack(Entity* self) {
+    WeaponAnimation* anim;
+    s8 animIndex;
+
+    self->posX.val = PLAYER.posX.val;
+    self->posY.val = PLAYER.posY.val;
+    self->facingLeft = PLAYER.facingLeft;
+    animIndex = (self->params & 0x7fff) >> 8;
+    anim = &D_27000_8017B2F4[animIndex];
+
+    if (PLAYER.ext.player.anim >= anim->frameStart &&
+        PLAYER.ext.player.anim < (anim->frameStart + 7) && g_Player.unk46) {
+        if (self->step == 0) {
+            SetSpriteBank1(g_Animset);
+            self->animSet = ANIMSET_OVL(0x10);
+            self->palette = PAL_DRA(0x110);
+            self->unk5A = 0x64;
+            if (g_HandId != 0) {
+                self->animSet += 2;
+                self->palette += 0x18;
+                self->unk5A += 2;
+            }
+            self->palette += anim->palette;
+            self->flags = FLAG_UNK_20000 | FLAG_UNK_40000;
+            self->zPriority = PLAYER.zPriority - 2;
+            self->drawMode = DRAW_TPAGE2 | DRAW_TPAGE;
+            SetWeaponProperties(self, 0);
+            self->step++;
+        }
+        self->ext.weapon.anim = PLAYER.ext.player.anim - anim->frameStart;
+        if ((PLAYER.animFrameDuration == 1) &&
+            (PLAYER.animFrameIdx == anim->soundFrame)) {
+            g_api.PlaySfx(anim->soundId);
+            g_api.PlaySfx(SFX_ANIME_SWORD_B);
+        }
+        if (g_api.UpdateUnarmedAnim(anim->frameProps, anim->frames) < 0) {
+            DestroyEntity(self);
+            return;
+        }
+    } else {
+        DestroyEntity(self);
+        return;
+    }
+
+    if ((PLAYER.animFrameIdx - 6) < 2U) {
```

Comment:
These can usually be rewritten as something like `if 5 < PLAYER.animFrameIdx && PLAYER.animFrameIdx < 7`, give something like that a try. Otherwise looks like a good PR, thanks!